### PR TITLE
upgrade stacks

### PR DESCRIPTION
**Pull request by lecorguille**
* [X] I have read the guidelines above.
* [ ] This PR adds a new recipe.
* [X] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by yhoogstrate** [comment]
`checking for functional regular expressions... configure: error: Regular expressions are not functional, you need g++ 4.9.0 or greater.`

**Comment by lecorguille** [comment]
Thanks @yhoogstrate for the info :/

File: recipes/stacks/1.45/build.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+export CPATH=${PREFIX}/include
+
+# Temporary fix for 1.42 https://groups.google.com/forum/#!topic/stacks-users/fLQ_qMhygQ4
+ sed -i.bak "41 c extern int encoded_gtypes[4][4];" src/populations.cc
```
Comment (abretaud) [inline recipes/stacks/1.45/build.sh]: Can you remove this fix? It's no longer needed now

**Comment by abretaud** [comment]
I just found the same problem while trying to do the same recipe :)

**Comment by abretaud** [comment]
ICYMI, there's a 1.46 version out now (though it will still fail due to gcc dep)
Also, I think the latest recipe version should be at the root of the stacks dir ([xref](https://bioconda.github.io/guidelines.html#versions))

**Comment by johanneskoester** [comment]
## Linting

The following problems have been found

| Recipe       | failed checks |
| ------------ | ------------- |
| recipes/mothur/1.36.1 | [missing_license](https://bioconda.github.io/linting.html#missing_license) || recipes/mothur/1.36.1 | [missing_summary](https://bioconda.github.io/linting.html#missing_summary) |

**Comment by johanneskoester** [comment]
That is actually not me but our new linting mechanism :-). I still have to find out why it lints mothur. It has not been changed in this PR.

**Comment by abretaud** [comment]
Green \o/

**Comment by lecorguille** [comment]
👏 Well done @abretaud

**Pull request merged**